gapic/src/main/com/google/gapid/perfetto/models/FrameInfo.java

```diff
@@ -31,70 +37,307 @@
 public class FrameInfo {
   public static final FrameInfo NONE = new FrameInfo(Collections.emptyList());
 
-  private static final String FRAME_SLICE_QUERY =
-      "select t.id, t.name, t.scope, max(depth) + 1 " +
-      "from gpu_track t join frame_slice s on (t.id = s.track_id) " +
-      "group by t.id " +
-      "order by t.id";
-  private static final String DISPLAYED_FRAME_TRACK_NAME = "Displayed Frame";
+  private static final String LAYER_NAME_QUERY =
+      "select distinct layer_name from frame_slice";
+  private static final String LAYER_TRACKS_QUERY =
+      "select name, track_id from gpu_track inner join " +
+      "(select distinct track_id from frame_slice where layer_name = '%s') t" +
+      " on gpu_track.id = t.track_id";
+  private static final String TRACK_ID_QUERY =
+      "select group_concat(track_id) from (%s) where name GLOB '%s'";
```

Comment:
These name globbing bothers me.  It's alright for this CL, but remind me to talk with you about this.